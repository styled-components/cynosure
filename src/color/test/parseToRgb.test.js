import parseToRgb from '../parseToRgb'

describe('parseToRgb', () => {
  it('should parse a hex color representation', () => {
    expect(parseToRgb('#Ff43AE')).toMatchSnapshot()
  })

  it('should parse an 8-digit hex color representation', () => {
    expect(parseToRgb('#Ff43AEFF')).toMatchSnapshot()
  })

  it('should parse an 4-digit hex color representation', () => {
    expect(parseToRgb('#0f08')).toMatchSnapshot()
  })

  it('should parse a reduced hex color representation', () => {
    expect(parseToRgb('#45a')).toMatchSnapshot()
  })

  it('should parse a rgba color representation', () => {
    expect(parseToRgb('rgba(174,67,255,0.6)')).toMatchSnapshot()
    expect(parseToRgb('rgba( 174 , 67 , 255 , 0.6 )')).toMatchSnapshot()
  })

  it('should parse a rgb color representation', () => {
    expect(parseToRgb('rgb(174,67,255)')).toMatchSnapshot()
    expect(parseToRgb('rgb( 174 , 67 , 255 )')).toMatchSnapshot()
  })

  it('should parse a hsl color representation', () => {
    expect(parseToRgb('hsl(210,10%,4%)')).toMatchSnapshot()
    expect(parseToRgb('hsl( 210 , 10% , 4% )')).toMatchSnapshot()
  })

  it('should parse a hsla color representation', () => {
    expect(parseToRgb('hsla(210,10%,40%,0.75)')).toMatchSnapshot()
    expect(parseToRgb('hsla( 210 , 10% , 40% , 0.75 )')).toMatchSnapshot()
  })

  it('should throw an error if an invalid color string is provided', () => {
    expect(() => {
      parseToRgb('(174,67,255)')
    }).toThrow(
      "Couldn't parse the color string. Please provide the color as a string in hex, rgb, rgba, hsl or hsla notation.",
    )
  })

  it('should throw an error if an invalid color string is provided', () => {
    expect(() => {
      parseToRgb(12345)
    }).toThrow(
      'Passed an incorrect argument to a color function, please pass a string representation of a color.',
    )
  })

  it('should parse a hsla color representation with decimal values', () => {
    expect(parseToRgb('hsla(210,0.5%,0.5%,1.0)')).toMatchSnapshot()
    expect(parseToRgb('hsla( 210 , 0.5% , 0.5% , 1.0 )')).toMatchSnapshot()
  })

  it('should throw an error if an invalid hsl string is provided', () => {
    expect(() => {
      parseToRgb('hsl(210,120%,4%)')
    }).toThrow(
      `Couldn't generate valid rgb string from ${'hsl(210,120%,4%)'}, it returned ${'rgb(-2,10,22)'}.`,
    )
  })

  it('should throw an error if an unparsable hsla string is provided', () => {
    expect(() => {
      parseToRgb('hsla(210,120%,4%,0.7)')
    }).toThrow(
      `Couldn't generate valid rgb string from ${'hsla(210,120%,4%,0.7)'}, it returned ${'rgb(-2,10,22)'}.`,
    )
  })
})
