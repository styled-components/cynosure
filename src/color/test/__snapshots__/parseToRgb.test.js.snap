// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseToRgb should parse a hex color representation 1`] = `
Object {
  "blue": 174,
  "green": 67,
  "red": 255,
}
`;

exports[`parseToRgb should parse a hsl color representation 1`] = `
Object {
  "blue": 11,
  "green": 10,
  "red": 9,
}
`;

exports[`parseToRgb should parse a hsl color representation 2`] = `
Object {
  "blue": 11,
  "green": 10,
  "red": 9,
}
`;

exports[`parseToRgb should parse a hsl color representation with decimal values 1`] = `
Object {
  "blue": 38,
  "green": 33,
  "red": 28,
}
`;

exports[`parseToRgb should parse a hsl color representation with decimal values 2`] = `
Object {
  "blue": 38,
  "green": 33,
  "red": 28,
}
`;

exports[`parseToRgb should parse a hsla color representation 1`] = `
Object {
  "alpha": 0.75,
  "blue": 112,
  "green": 102,
  "red": 92,
}
`;

exports[`parseToRgb should parse a hsla color representation 2`] = `
Object {
  "alpha": 0.75,
  "blue": 112,
  "green": 102,
  "red": 92,
}
`;

exports[`parseToRgb should parse a hsla color representation with decimal values 1`] = `
Object {
  "alpha": 1,
  "blue": 0,
  "green": 0,
  "red": 0,
}
`;

exports[`parseToRgb should parse a hsla color representation with decimal values 2`] = `
Object {
  "alpha": 1,
  "blue": 0,
  "green": 0,
  "red": 0,
}
`;

exports[`parseToRgb should parse a reduced hex color representation 1`] = `
Object {
  "blue": 170,
  "green": 85,
  "red": 68,
}
`;

exports[`parseToRgb should parse a rgb color representation 1`] = `
Object {
  "blue": 255,
  "green": 67,
  "red": 174,
}
`;

exports[`parseToRgb should parse a rgb color representation 2`] = `
Object {
  "blue": 255,
  "green": 67,
  "red": 174,
}
`;

exports[`parseToRgb should parse a rgba color representation 1`] = `
Object {
  "alpha": 0.6,
  "blue": 255,
  "green": 67,
  "red": 174,
}
`;

exports[`parseToRgb should parse a rgba color representation 2`] = `
Object {
  "alpha": 0.6,
  "blue": 255,
  "green": 67,
  "red": 174,
}
`;

exports[`parseToRgb should parse an 4-digit hex color representation 1`] = `
Object {
  "alpha": 0.53,
  "blue": 0,
  "green": 255,
  "red": 0,
}
`;

exports[`parseToRgb should parse an 8-digit hex color representation 1`] = `
Object {
  "alpha": 1,
  "blue": 174,
  "green": 67,
  "red": 255,
}
`;
